src/Sasl/Mechanism.php

```diff
@@ -0,0 +1,67 @@
+<?php
+/* vim: set expandtab tabstop=4 shiftwidth=4 softtabstop=4 foldmethod=marker: */
+// +---------------------------------------------------------------------------
+// | SWAN [ $_SWANBR_SLOGAN_$ ]
+// +---------------------------------------------------------------------------
+// | Copyright $_SWANBR_COPYRIGHT_$
+// +---------------------------------------------------------------------------
+// | Version  $_SWANBR_VERSION_$
+// +---------------------------------------------------------------------------
+// | Licensed ( $_SWANBR_LICENSED_URL_$ )
+// +---------------------------------------------------------------------------
+// | $_SWANBR_WEB_DOMAIN_$
+// +---------------------------------------------------------------------------
+
+namespace Kafka\Sasl;
+
+use Kafka\CommonSocket;
+use Kafka\Exception;
+use Kafka\Protocol;
+use Kafka\Protocol\Protocol as ProtocolTool;
+
+/**
++------------------------------------------------------------------------------
+* Kafka sasl provider mechanism abstract
++------------------------------------------------------------------------------
+*
+* @package
+* @version $_SWANBR_VERSION_$
+* @copyright Copyleft
+* @author $_SWANBR_AUTHOR_$
++------------------------------------------------------------------------------
+*/
+
+abstract class Mechanism
+{
+    // {{{ consts
+    // }}}
+    // {{{ functions
+    // {{{ public function handShake()
+    
+    /**
+     *
+     * sasl authenticate hand shake
+     *
+     * @access public
+     * @return void
+     */
+    protected function handShake(CommonSocket $socket, string $mechanism)
```

Comment:
This should be `private` (considering the other suggestion I've gave)

Reply:
And it's missing `: void`

Reply:
This should be `protected`, child class will call this